armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -386,6 +446,169 @@ def axiallyExpandAssembly(self):
         bounds[2] = array(mesh)
         self.linked.a.spatialGrid._bounds = tuple(bounds)
 
+    def _shiftLinkedCompsForDelta(self, c: "Component", cAbove: "Component", deltaZTop: float):
+        # shift the height and ztop of c downwards (-deltaZTop) or upwards (+deltaZTop)
+        c.height += deltaZTop
+        c.ztop += deltaZTop
+        # the height of cAbove grows and zbottom moves downwards (-deltaZTop) or shrinks and moves upward (+deltaZTop)
+        cAbove.height -= deltaZTop
+        cAbove.zbottom += deltaZTop
+
+    def redistributeMass(self, fromComp: "Component", toComp: "Component", deltaZTop: float):
```

Comment:
I won't die on this hill, but this method feels too long. It would be ideal if it could be broken up into 2 to 5 little helper methods that this one calls. Just to break up the logic and make it more readable.